src/Admin/BlockAdmin.php

```diff
@@ -245,4 +215,65 @@ private function getDefaultTemplate(BlockServiceInterface $blockService)
             return $options['template'];
         }
     }
+
+    private function configureBlockFields(FormMapper $formMapper, BlockInterface $block): void
+    {
+        $service = $this->blockManager->get($block);
+        $blockType = $block->getType();
+
+        if (!$service instanceof BlockServiceInterface) {
+            throw new InvalidArgumentException(sprintf('The block "%s" is not a valid %s', $block->getType(), BlockServiceInterface::class));
+        }
+
+        if ($service instanceof EditableBlockService) {
+            $blockMapper = new PageFormMapper($formMapper);
+            if ($block->getId() > 0) {
+                $service->configureEditForm($blockMapper, $block);
+            } else {
+                $service->configureCreateForm($blockMapper, $block);
+            }
+        } else {
+            @trigger_error(
+                sprintf(
+                    'Editing a non %s block service is deprecated since sonata-project/page-bundle 3.x and will be removed with version 4.0.',
+                    EditableBlockService::class
+                ),
+                E_USER_DEPRECATED
+            );
+
+            if ($block->getId() > 0) {
+                $service->buildEditForm($formMapper, $block);
+            } else {
+                $service->buildCreateForm($formMapper, $block);
+            }
+        }
+
+        if ($formMapper->has('settings') && isset($this->blocks[$blockType]['templates'])) {
+            $settingsField = $formMapper->get('settings');
+
+            if (!$settingsField->has('template')) {
+                $choices = [];
+
+                if (null !== $defaultTemplate = $this->getDefaultTemplate($service)) {
+                    $choices['default'] = $defaultTemplate;
+                }
+
+                foreach ($this->blocks[$blockType]['templates'] as $item) {
+                    $choices[$item['name']] = $item['template'];
+                }
+
+                if (\count($choices) > 1) {
+                    $templateOptions = [
+                        'choices' => $choices,
+                    ];
+
+                    if ($settingsField->hasOption('choices_as_values')) {
```

Comment:
?!

AFAIK this is just a hack symfony 2

Reply:
Indeed, but it can be passed with `false` as value: https://symfony.com/doc/2.7/reference/forms/types/choice.html#choices-as-values
>The choices_as_values option was added to keep backward compatibility with the old way of handling the choices option. When set to false (or omitted), the choice keys are used as the underlying value and the choice values are shown to the user.

Reply:
I would leave is as it is, I just moved existing code to a new method and not related to this PR.

Btw. this setting does not exist in symfony 4 and was deprecated in symfony 3: https://symfony.com/doc/3.3/reference/forms/types/choice.html#choices-as-values

Reply:
If this check was already part of the codebase, it's fine for me.